File: flink-demo/flinkScalaCode/src/main/scala/com/yw/flink/example/scalacases/case01_source/Case01_FileSource.scala
package com.yw.flink.example.scalacases.case01_source

import org.apache.flink.api.common.eventtime.WatermarkStrategy
import org.apache.flink.connector.file.src.FileSource
import org.apache.flink.connector.file.src.reader.TextLineInputFormat
import org.apache.flink.core.fs.Path
import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment

/**
  * Flink Scala 读取File 文件数据
  */
object Case01_FileSource {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    //导入隐式转换
    import org.apache.flink.streaming.api.scala._

    val fileSource: FileSource[String] = FileSource.forRecordStreamFormat(new TextLineInputFormat(),
      new Path("hdfs://mycluster/flinkdata/a.txt")).build()

    val result: DataStream[String] = env.fromSource(fileSource, WatermarkStrategy.noWatermarks(), "file-source")
    result.print()
    env.execute()
  }

}

File: flink-demo/flinkJavaCode/src/main/java/com/yw/flink/example/javacases/case01_source/Case01_FileSource.java
package com.yw.flink.example.javacases.case01_source;

import org.apache.flink.api.common.eventtime.WatermarkStrategy;
import org.apache.flink.connector.file.src.FileSource;
import org.apache.flink.connector.file.src.reader.TextLineInputFormat;
import org.apache.flink.core.fs.Path;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

/**
 * Flink 读取HDFS 文件数据
 */
public class Case01_FileSource {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
//        DataStreamSource<String> result = env.readTextFile("hdfs://mycluster/flinkdata/a.txt");
        FileSource<String> fileSource = FileSource.forRecordStreamFormat(new TextLineInputFormat(), new Path("hdfs://mycluster/flinkdata/a.txt")).build();

        //使用source
        DataStreamSource<String> result = env.fromSource(fileSource, WatermarkStrategy.noWatermarks(), "file-source");

        result.print();
        env.execute();
    }
}
